accepted/install-locations.md

```diff
@@ -0,0 +1,227 @@
+
+## .NET Core install layout
+The content of a .NET Core install location always follows the same layout. The directory structure is the same on all platforms, file names may differ to conform with the platform specific requirements. (Below sample uses file names for Windows):
+* `./dotnet.exe` - the "muxer" which is used to execute apps and issue dotnet CLI commands.
+* `./host/fxr/<version>/hostfxr.dll` - this library contains all the logic to resolve framework references and select SDK. Regardless of application, the latest available version is used.
+* `./shared` - contains shared frameworks.
+  * `./Microsoft.NETCore.App/<version>` - the root framework which contains the .NET Core runtime. This framework contains `hostpolicy.dll` which is the entry point called by the `hostfxr` to run the app. Applications can choose which version to use.
+  * `./<FrameworkName>/<version>` - any other shared framework. Applications can choose which version to use.
+* `./sdk/<version>` - only installed when .NET Core SDK is installed - contains the SDK tools. Users can choose which version to use.
+
+
+## Installation types
+.NET Core can be installed in several ways, each provides a different location and expected behavior. Each such location will contain the .NET Core install layout described above.
+
+### Global install to default location
+This is the most common case where an installer installs the runtime (and potentially SDK) to a global default location accessible to everything on the system (and by all users).
+* Windows
+  * 32-bit on 64-bit OS - `%ProgramFiles(x86)%\dotnet`
+  * Otherwise - `%ProgramFiles%\dotnet`
+* Linux - the host searches in `/usr/share/dotnet` as a fallback, but the installer may pick a different default based on the requirements of the specific distro/configuration. In such case the install is effectively just like the global install to custom location described below. If correctly registered the host will search the custom location first.
+* macOS - `/usr/local/share/dotnet`
+
+*This type of install is supported by .NET Core 2.2 and earlier.*
+
+### Global install to custom location
+.NET Core is installed globally for the machine but into a non-default (custom) location. The location of the install is stored in platform specific configuration so that it can be found by everything on the system (and by all users).
+
+This type of install should effectively behave the same as install to the default location. Can be used to put .NET Core on a different drive to save space on the system drive, or to serve other purposes.
+
+*This type of install will for sure be supported by .NET Core 3.0 on Windows, other cases are not yet committed.*
+
+### Per-user install
+.NET Core installation which is per-user, thus doesn't require admin/root access to the machine. Otherwise it should behave just like a global install, but only accessible to the user which installed it.
```

Comment:
Why is this scenario any different than xcopy deployed .NET Core?

Reply:
The idea was that it would act like the global install, just per-user. If I would install per-user I could use it just like the global install today. But it would not require admin rights to install.

Given the feedback, I'll probably remove this from the proposal. It's true that we don't currently have a scenario which would require it.

Reply:
This scenario is actually one that people request a lot and was originally supported by original MLL plus default dotnet-install script location. They want to be able to pull down a CLI without elevating and have it get picked up by IDE. I went into more detail on another comment. Every time it comes up, it gets shot down, but we should not hand wave away that current xcopy story is equivalent. People ask for something better all the time. Nobody likes setting environment variables to get their private copy working in the IDE. It's razzle all over again.